File: thr.py
################################################
#
# COMS30068 - thr.py
# TOPIC: create, save and display an image
#
# Getting-Started-File for OpenCV
# University of Bristol
#
################################################

import cv2
import numpy as np

# Read image from file
image = cv2.imread("mandrill.jpg", 1)

# Convert to grey scale
gray_image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY )

# Create mask from threshold 
th = 128
gray_image = gray_image>128
gray_image = gray_image*255

# Or threshold by looping through all pixels
# for y in range(0, image.shape[0]):  # go through all rows (or scanlines)
#	for x in range(0, image.shape[1]):  # go through all columns
#		if gray_image[y, x] > 128:
#			gray_image[y, x] = 255
#		else:
#			gray_image[y, x] = 0


# Save thresholded image
cv2.imwrite("thr.jpg", gray_image)

File: cpp/thr.cpp
/////////////////////////////////////////////////////////////////////////////
//
// COMS30121 - thr.cpp
// TOPIC: basic thresholding
//
// Getting-Started-File for OpenCV
// University of Bristol
//
/////////////////////////////////////////////////////////////////////////////

#include <stdio.h>
#include <opencv/cv.h>        //you may need to
#include <opencv/highgui.h>   //adjust import locations
#include <opencv/cxcore.h>    //depending on your machine setup

using namespace cv;

int main() { 

  // Read image from file
  Mat image = imread("mandrill.jpg", 1);

  // Convert to grey scale
  Mat gray_image;
  cvtColor(image, gray_image, CV_BGR2GRAY);

  // Threshold by looping through all pixels
  for (int y = 0; y<gray_image.rows; y++) {
    for (int x = 0; x<gray_image.cols; x++) {
      uchar pixel = gray_image.at<uchar>(y, x);
      if (pixel>128) gray_image.at<uchar>(y, x) = 255;
      else gray_image.at<uchar>(y, x) = 0;
  } }

  //Save thresholded image
  imwrite("thr.jpg", gray_image);

  return 0;
}